### PR TITLE
Add icon component

### PR DESCRIPTION
**Pull request by justinvdm**

### CODE REVIEW

**Comment by justinvdm** [comment]
@peterp depends on #55, otherwise ready for review.

File: src/components/HeaderIcon/__tests__/HeaderIcon-test.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+
+describe('HeaderIcon', () => {
+  it('should render', () => {
+    expect(render(
+      <Header theme={Header.themes.dark}>
+        <HeaderIcon type={HeaderIcon.types.backLight} />
+      </Header>
+    )).toMatchSnapshot();
+  });
```
Comment (justinvdm) [inline src/components/HeaderIcon/__tests__/HeaderIcon-test.js]: screen shot 2016-08-24 at 1 39 53 pm

File: src/components/HeaderIcon/__tests__/HeaderIcon-test.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+  it('should support style overriding', () => {
+    expect(render(
+      <Header theme={Header.themes.dark}>
+        <HeaderIcon type={HeaderIcon.types.backLight} style={{ left: 48 }} />
+      </Header>
+    )).toMatchSnapshot();
```
Comment (justinvdm) [inline src/components/HeaderIcon/__tests__/HeaderIcon-test.js]: screen shot 2016-08-24 at 1 40 30 pm

Comment (justinvdm) [inline src/components/HeaderIcon/__tests__/HeaderIcon-test.js]: screen shot 2016-08-24 at 2 13 30 pm

File: src/components/Icon/__tests__/Icon-test.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+import Icon from 'src/components/Icon';
+
+
+describe('Icon', () => {
+  it('should render its `backLight` type', () => {
+    expect(render(
+      <Icon type={Icon.types.backLight} />
+    )).toMatchSnapshot();
+  });
```
Comment (justinvdm) [inline src/components/Icon/__tests__/Icon-test.js]: screen shot 2016-08-24 at 1 39 53 pm

File: src/components/Icon/__tests__/Icon-test.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+  it('should support style overriding', () => {
+    expect(render(
+      <Icon
+        type={Icon.types.backLight}
+        style={{ backgroundColor: '#333' }}
+      />
+    )).toMatchSnapshot();
```
Comment (justinvdm) [inline src/components/Icon/__tests__/Icon-test.js]: screen shot 2016-08-24 at 1 40 30 pm

File: src/components/HeaderIcon/__tests__/HeaderIcon-test.js (hunk 1 above)
Comment (justinvdm) [inline src/components/HeaderIcon/__tests__/HeaderIcon-test.js]: screen shot 2016-08-24 at 2 35 25 pm

File: src/components/Icon/index.js [hunk 5]
```diff
@@ -0,0 +1,28 @@
+import React, { PropTypes } from 'react';
+import { Image } from 'react-native';
+
+import backLight from 'app/assets/back-light.png';
```
Comment (peterp) [inline src/components/Icon/index.js]: I've created an `constants/images`

**Comment by peterp** [comment]
Besides the `src/constansts/images`, 👍

File: src/components/Icon/index.js (hunk 5 above)
Comment (justinvdm) [inline src/components/Icon/index.js]: 🍰

**Pull request merged**